webui/react/src/pages/WorkspaceDetails/WorkspaceQuickSearch.tsx

```diff
@@ -26,11 +29,14 @@ const WorkspaceQuickSearch: React.FC<Props> = ({ children }: Props) => {
   const [workspaceMap, setWorkspaceMap] = useState<Map<Workspace, Project[]>>(new Map());
   const [isLoading, setIsLoading] = useState(true);
   const [isModalVisible, setIsModalVisible] = useState(false);
+  const workspaceObservable = useObservable(workspaceStore.workspaces);
+  const workspaces = Loadable.getOrElse([], workspaceObservable);
 
   const fetchData = useCallback(async () => {
     try {
-      const workspaceResponse = await getWorkspaces({ limit: 0, sortBy: 'SORT_BY_NAME' });
-      const filteredWorkspaces = workspaceResponse.workspaces.filter((w) => !w.immutable);
+      const filteredWorkspaces = workspaces
+        .filter((w) => !w.immutable)
+        .sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1));
```

Comment:
**suggestion:** my thought is that we can move this into the workspace store itself? Since we have `pinned` and `unarchived` already?

Reply:
to clarify, create a shortcut called `.sorted` and replace anywhere else (I don't know for sure if we have other examples) with alphabetically sorted workspaces?  (note that pinned workspaces uses created time for sort)

Reply:
oops I meant to highlight lines 37-39. I meant the whole filter and sort could be something that is done in the store and then returned as an observable called `immutables` or something. Same pattern as `unarchived` for example:
```
  public readonly unarchived = this.#loadableWorkspaces.select((loadable) => {
    return Loadable.quickMatch(loadable, NotLoaded, (workspaces) => {
      return Loaded(workspaces.filter((workspace) => !workspace.archived));
    });
  });
```
Where once the workspaces are loaded the logic to provide a pre-canned set of workspaces is all done in the store.